lib/coffee_place/result.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+module CoffeePlace
+  Result = Struct.new(:success?, :value, :error) do
```

Comment:
How do you decide when to return instances of this class from a method? Seeing it is used across the solution.

Reply:
In Ruby, predicate methods return `true` / `false`, while finder methods return `some truthy value` / nil.
I find that in a lot of situations a `Result` type, with two variants - `Success(value)`, and `Failure(error)` comes in handy. This is common in many functional languages, but also useful in OOP, when the return value you need
has to be able to represent different variants.

A really nice gem for handling this across an application is [dry-monads](https://dry-rb.org/gems/dry-monads/1.3/)
You don't have to go heavy on the FP. Just using variant types consistently, when neither a predicate or finder
do the job is sufficient.

This also enables neat ways of assembling operations, as seen in [dry-transaction](https://dry-rb.org/gems/dry-transaction/0.13/)
Note: `dry-transaction` seems to be undergoing API changes, as it is not in a `1.0` version yet :D

I've used this gem in the past in a production application, validating user submitted Excel files, and it worked pretty nicely - the code was easy to extend and refactor.